sirepo/job_driver/__init__.py

```diff
@@ -163,27 +187,23 @@ def make_lib_dir_symlink(self, op):
         """Only one op of each type allowed
 
         """
+        n = op.opName
+        if n in (job.OP_CANCEL, job.OP_KILL):
+            return
+        if n == job.OP_SBATCH_LOGIN:
+            l = [o for o in self.ops.values() if o.opId != op.opId]
+            assert not l, \
+                'received {} but have other ops={}'.format(op, l)
+            return
+        if op.op_slot:
+            return
+        q = self.op_q[n]
         try:
-            n = op.opName
-            if n in (job.OP_CANCEL, job.OP_KILL):
-                return
-            if n == job.OP_SBATCH_LOGIN:
-                l = [o for o in self.ops.values() if o.opId != op.opId]
-                assert not l, \
-                    'received {} but have other ops={}'.format(op, l)
-                return
-            if op.op_slot:
-                return
-            q = self.op_q[n]
-            try:
-                op.op_slot = q.get_nowait()
-            except tornado.queues.QueueEmpty:
-                pkdlog('self={} op={} await op_q.get()', self, op)
-                op.op_slot = await q.get()
-                raise job_supervisor.Awaited()
-        finally:
-            # may be redundant, but need to track here
-            self.ops[op.opId] = op
+            op.op_slot = q.get_nowait()
+        except tornado.queues.QueueEmpty:
```

Comment:
The reason this was narrow is that it isolates the QueueEmpty to the function which could get that exception. Too wide a catch is a problem when catching a specific exception like this.

Reply:
I believe it still does just that. I removed the try/finally but didn't change this try/except.